cmd/zed/agents/zfs_retire.c

```diff
@@ -268,16 +324,63 @@ replace_with_spare(fmd_hdl_t *hdl, zpool_handle_t *zhp, nvlist_t *vdev)
 		fmd_hdl_debug(hdl, "zpool_vdev_replace '%s' with spare '%s'",
 		    dev_name, basename(spare_name));
 
-		if (zpool_vdev_attach(zhp, dev_name, spare_name,
-		    replacement, B_TRUE) == 0)
+		ret = zpool_vdev_attach(zhp, dev_name, spare_name,
+		    replacement, B_TRUE);
+		if (ret == 0)
+			break;
+
+		if (strstr(libzfs_error_description(zhdl), "busy") != NULL) {
```

Comment:
It would be reasonable to change `zpool_vdev_attach` to return a new `EZFS_BUSY` in the `EBUSY` case (instead of EZFS_BADDEV), then update this check to `if (zpool_get_handle(zhp)->libzfs_error == EZFS_BUSY) { }`.